ts/config.ts

```diff
@@ -73,11 +93,116 @@ export const nody_greeter: nody_config = {
     theme_dir:
       process.env.NODY_GREETER_THEME_DIR || "/usr/share/web-greeter/themes/",
   },
+  theme: {
+    primary_html: "index.html",
+    secondary_html: "",
+  },
 };
 
 const path_to_config =
   process.env.NODY_GREETER_CONFIG || "/etc/lightdm/web-greeter.yml";
 
+let theme_dir: string | undefined;
+
+export function load_theme_dir(): string {
+  const theme = nody_greeter.config.greeter.theme;
+  const dir = nody_greeter.app.theme_dir;
+  const def_theme = "gruvbox";
+  theme_dir = path.join(dir, theme);
+
+  if (theme.startsWith("/")) theme_dir = theme;
+  else if (theme.includes(".") || theme.includes("/"))
+    theme_dir = path.join(process.cwd(), theme);
+
+  if (theme_dir.endsWith(".html")) theme_dir = path.dirname(theme_dir);
+
+  if (!fs.existsSync(theme_dir)) {
```

Comment:
I've just realised thata we're using synchronous / blocking filesystem operations throughout here. It would be better to use async functions, and use await [`fs.promises`](https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api), so that we don't freeze the Node.js process.